src/libinjector/linux/linux_utils.c

```diff
@@ -0,0 +1,206 @@
+    if (!drakvuf_get_kernel_struct_member_rva(drakvuf, "mm_struct", "vdso", &offset))
+    {
+        PRINT_DEBUG("Failed to get vdso offset\n");
+        drakvuf_release_vmi(drakvuf);
+        return 0;
+    }
+
+    PRINT_DEBUG("vdso offset: %ld\n", offset);
+    ctx.addr = addr + offset;
+
+    // since vdso is a pointer
+    if (VMI_SUCCESS != vmi_read_64(vmi, &ctx, &addr))
+    {
+        PRINT_DEBUG("Failed to read vdso address\n");
+        drakvuf_release_vmi(drakvuf);
+        return 0;
+    }
+
+    PRINT_DEBUG("Got vdso address: %lx\n", addr);
+    drakvuf_release_vmi(drakvuf);
+
+    return addr;
+}
+
+/** src: https://www.geeksforgeeks.org/naive-algorithm-for-pattern-searching/ **/
+static size_t search(char* txt, char* pat, int N, int M)
+{
+    /* A loop to slide pat[] one by one */
+    for (int i = 0; i <= N - M; i++)
+    {
+        int j;
+
+        /* For current index i, check for pattern match */
+        for (j = 0; j < M; j++)
+            if (txt[i + j] != pat[j])
+                break;
+
+        if (j == M) // if pat[0...M-1] = txt[i, i+1, ...i+M-1]
+            return i;
+    }
+    return -1;
+}
+
+addr_t find_syscall(drakvuf_t drakvuf, drakvuf_trap_info_t* info, addr_t vdso)
+{
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    ACCESS_CONTEXT(ctx,
+        .translate_mechanism = VMI_TM_PROCESS_PID,
+        .pid = info->proc_data.pid,
+        .addr = vdso
+    );
+
+    size_t size = 4096;
+    size_t bytes_read = 0;
+    char* vdso_memory = g_try_malloc(size);
+
+    // read the vdso memory
+    if (VMI_SUCCESS != vmi_read(vmi, &ctx, size, (void*)vdso_memory, &bytes_read))
+    {
+        fprintf(stderr, "Could not read vdso memory\n");
+        g_free(vdso_memory);
+        return 0;
+    }
+
+    PRINT_DEBUG("vdso memory read successful\n");
+    drakvuf_release_vmi(drakvuf);
+
+    char syscall[] = { 0xf, 0x5 };
+    int syscall_offset = search(vdso_memory, syscall, size, 2);
+    if (syscall_offset == -1)
+    {
+        PRINT_DEBUG("Failed to get syscall offset\n");
+        g_free(vdso_memory);
+        return 0;
+    }
+    PRINT_DEBUG("syscall offset: %d\n", syscall_offset);
+    PRINT_DEBUG("syscall addr: %lx\n", vdso + syscall_offset);
+
+    free(vdso_memory);
+
+    injector_t injector = info->trap->data;
+    injector->syscall_addr = vdso + syscall_offset;
+
+    return vdso + syscall_offset;
+}
+
+bool setup_post_syscall_trap(drakvuf_t drakvuf, drakvuf_trap_info_t* info, addr_t syscall_addr)
+{
+    injector_t injector = info->trap->data;
+
+    injector->bp = g_try_malloc0(sizeof(drakvuf_trap_t));
+
+    injector->bp->type = BREAKPOINT;
+    injector->bp->name = "injector_post_syscall_trap";
+    // cb will be set from previous call only
+    // we don't have injector_int3_userspace_cb function
+    // in scope here so we will use it from the previous trap
+    injector->bp->cb = info->trap->cb; //injector_int3_userspace_cb;
```

Comment:
maybe remove static from injector_int3_userspace_cb if it's required in other parts of the code?